### DISCUSSION TITLE
We built a Google Shopping integration

### DISCUSSION THREAD

**Discussion by mewa** [swellstores/community#154] [Show and tell]
Hi all,

We've been using Swell to run our stores and while it's a great platform we missed the ability to list your inventory on Google Shopping. So we wrote our own integration which we're now happy to share with you!

With [Swell Toolkit](https://swelltoolkit.com?utm_source=github&utm_medium=announcement&utm_content=link) you can effortlessly (2 min setup) plug in your Swell store's inventory into Google Merchant Center and drive traffic from free listings as well as paid traffic via Google Ads.

Here's a short demo ([link](https://swelltoolkit.com/demo?utm_source=github&utm_medium=announcement&utm_content=demo)):

[![Watch demo](https://cdn.loom.com/sessions/thumbnails/8c377fdc296a4c4ca87ed9904348397a-1677151765516-with-play.gif)](https://swelltoolkit.com/demo?utm_source=github&utm_medium=announcement&utm_content=video)

Would love to hear your thoughts!

Thanks,
Marcin

PS What other channels would you like to see (e.g. we already have Instagram on our roadmap)?

**Comment by ericingram** [comment]
This is awesome! Although the demo doesn't show up for me.

cc @StevenSwell @ana-estrela

**Reply by mewa** [reply]
Thanks! I couldn't embed the video here since it's on Loom, but here's a link directly to the demo: [swelltoolkit.com/demo](https://swelltoolkit.com/demo?utm_source=github&utm_medium=announcement&utm_content=comment)

**Comment by greph** [comment]
@mewa , very impressive!

I see that private keys are needed in order to access Swell store data. I'm curious: if specific Swell store application data were instead made available (scopes), which ones would be required to run the app? For example, 'read products', 'write to inventory', etc?

**Reply by mewa** [reply]
Cheers!

Read access to products, categories and attributes is all that's required.

I have some ideas with respect to feed optimization that might require a bit more in the future – but for now that's it.
